pkg/agent/flowexporter/connections/conntrack_test.go

```diff
@@ -164,22 +163,8 @@ func TestConnTrackOvsAppCtl_DumpFlows(t *testing.T) {
 	assert.Equal(t, len(outputFlow), totalConns, "Number of connections in conntrack table should be equal to outputFlow")
 }
 
-func TestSetupConnTrackParameters(t *testing.T) {
-	err := setupConntrackParameters()
-	if err != nil {
-		t.Skipf("Skipping test as trying to set up conntrack parameters returned an error: %v", err)
-	} else {
-		conntrackAcct, err := sysctl.GetSysctlNet("netfilter/nf_conntrack_acct")
-		require.NoError(t, err, "Cannot read nf_conntrack_acct")
-		assert.Equal(t, 1, conntrackAcct, "net.netfilter.nf_conntrack_acct value should be 1")
-		conntrackTimestamping, err := sysctl.GetSysctlNet("netfilter/nf_conntrack_timestamp")
-		require.NoError(t, err, "Cannot read nf_conntrack_timestamp")
-		assert.Equal(t, 1, conntrackTimestamping, "net.netfilter.nf_conntrack_timestamp value should be 1")
-	}
-}
-
 func TestConnTrackSystem_GetMaxConnections(t *testing.T) {
-	connDumperDPSystem := NewConnTrackSystem(&config.NodeConfig{}, &net.IPNet{})
+	connDumperDPSystem := NewConnTrackSystem(&config.NodeConfig{}, &net.IPNet{}, nil)
 	maxConns, err := connDumperDPSystem.GetMaxConnections()
 	assert.NoErrorf(t, err, "GetMaxConnections function returned error: %v", err)
 	expMaxConns, err := sysctl.GetSysctlNet("nf_conntrack_max")
```

Comment:
Thanks for taking care of TestSetupConnTrackParameters.
Will this work if this unit test runs on windows or mac os?

Reply:
Just tried on mac os, unfortunately TestSetupConnTrackParameters doesn't work:
```
>> go test -v --run TestSetupConnTrackParameters
# github.com/vmware-tanzu/antrea/pkg/agent/proxy
../../../pkg/agent/proxy/proxier.go:191:16: p.installLoadBalancerServiceFlows undefined (type *proxier has no field or method installLoadBalancerServiceFlows)
FAIL	github.com/vmware-tanzu/antrea/test/integration/agent [build failed]
```

Reply:
You should use `make docker-test-integration` to run integration test.

Reply:
Thanks, integration test passed on mac os.

Reply:
Proposed a new issue corresponding to the unit test failure on macOS: #1332

Reply:
Currently we only support running unit tests natively on Linux. You can run the tests on macOS using the `docker-test-unit` make target. We can try supporting running unit tests natively on macOS, but I'm not sure this would be very useful or would provide good coverage anyway.

I think the one thing that matters is that unit tests don't require root privileges and getting rid of the earlier hack to handle errors when setting kernel parameters is a good thing too.

Reply:
Using `docker-test-unit` target makes sense for macOS for dev purposes. I was not aware of this target. Thanks.
Here the attempt was to make unit tests platform-independent.
@dreamtalen I think we should not worry about providing support to run unit tests on any platform natively but rather on removing the dependency of any specific platform and root privileges. Please make sure the above make target runs successfully on macOS.

Moving the TestSetupConnTrackParameters to the integration test removes the dependency on the root privileges. In addition, changes were already made to the test to remove the dependency on objects created for Linux or Windows platform.

Reply:
To clarify, I think it's ok to have some unit tests that will only run on linux (by adding `_linux` to the file name or adding a linux build tag), but they should not rely on specific privileges to be run